### PR TITLE
Fix and Reenable `Style/SafeNavigation`

### PR DESCRIPTION
**Pull request by Hamms**
This rule encourages use of safe `&.` navigation over verbose presence checking. There are quite a few violations so I think it's worth a PR of its own, but my expectation is that the code changes themselves are safe.

## Links

https://www.rubydoc.info/gems/rubocop/0.43.0/RuboCop/Cop/Style/SafeNavigation

## Testing story

Relying on existing tests.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/passwords_controller.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def create
         return
       end
     end
-    unless verify_recaptcha || (current_user && current_user.admin?)
+    unless verify_recaptcha || (current_user&.admin?)
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/passwords_controller.rb]: nit: the parentheses aren't needed anymore

File: dashboard/app/controllers/report_abuse_controller.rb [hunk 2]
```diff
@@ -104,9 +104,9 @@ def report_abuse
 
   def report_abuse_form
     @react_props = {
-      name: (current_user.name unless current_user.nil?),
-      email: (current_user.email unless current_user.nil?),
-      age: (current_user.age unless current_user.nil?),
+      name: (current_user&.name),
+      email: (current_user&.email),
+      age: (current_user&.age),
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/report_abuse_controller.rb]: nit: the parentheses aren't needed anymore

File: dashboard/app/helpers/levels_helper.rb [hunk 3]
```diff
@@ -528,7 +528,7 @@ def non_blockly_puzzle_options
       sublevelCallback: @sublevel_callback,
     }
 
-    if (@game && @game.owns_footer_for_share?) || @legacy_share_style
+    if (@game&.owns_footer_for_share?) || @legacy_share_style
```
Comment (bethanyaconnor) [inline dashboard/app/helpers/levels_helper.rb]: nit: the parentheses aren't needed anymore

File: dashboard/app/helpers/levels_helper.rb [hunk 4]
```diff
@@ -666,12 +666,12 @@ def blockly_options
     end
 
     # User/session-dependent options
-    app_options[:disableSocialShare] = true if (current_user && current_user.under_13?) || app_options[:embed]
+    app_options[:disableSocialShare] = true if (current_user&.under_13?) || app_options[:embed]
```
Comment (bethanyaconnor) [inline dashboard/app/helpers/levels_helper.rb]: nit: the parentheses aren't needed anymore (or could cover both conditions)

File: dashboard/app/helpers/levels_helper.rb [hunk 5]
```diff
@@ -711,7 +711,7 @@ def blockly_options
     end
     app_options[:send_to_phone_url] = send_to_phone_url if app_options[:isUS]
 
-    if (@game && @game.owns_footer_for_share?) || @legacy_share_style
+    if (@game&.owns_footer_for_share?) || @legacy_share_style
```
Comment (bethanyaconnor) [inline dashboard/app/helpers/levels_helper.rb]: nit: the parentheses aren't needed anymore

**Review by bethanyaconnor — APPROVED** [review]
a handful of nits about parentheses but otherwise looks good

File: dashboard/app/controllers/passwords_controller.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/app/controllers/passwords_controller.rb]: Oh, good catches, thanks! I think I know what rule I'm gonna address next

https://github.com/code-dot-org/code-dot-org/blob/58f62036d1e7caecd9410fb245b7a038aa464051/.rubocop_todo.yml#L696-L699

Comment (bethanyaconnor) [reply dashboard/app/controllers/passwords_controller.rb]: Sounds good! Fixing them as a follow up sounds good to me

**Pull request merged**